app/modules/import-export/assets/js/pages/import/import-process/import-process.js

```diff
@@ -136,13 +150,17 @@ export default function ImportProcess() {
 	useEffect( () => {
 		if ( KIT_STATUS_MAP.INITIAL !== kitState.status || ( isResolvedData && 'apply-all' === importContext.data.actionType ) ) {
 			if ( importedData ) { // After kit upload.
+				if ( returnTo && isValidRedirectUrl( decodeURIComponent( returnTo ) ) ) {
+					window.location.href = decodeURIComponent( returnTo );
+					return;
+				}
 				navigate( '/import/complete' );
 			} else if ( 'apply-all' === importContext.data.actionType ) { // Forcing apply-all kit content.
 				if ( kitState.data?.manifest?.plugins || importContext.data.uploadedData?.manifest.plugins ) {
 					importContext.dispatch( { type: 'SET_PLUGINS_STATE', payload: 'have' } );
 				}
 				if ( uploadedData.conflicts && Object.keys( uploadedData.conflicts ).length && ! isResolvedData ) {
-					navigate( '/import/resolver' );
+					navigate( '/import/resolver?return_to=' + returnTo );
```

Comment:
@nuritsha @nicoladj77 Will this add the param even if `returnTo` is an empty string?

Reply:
@louiswol94 refactored